packages/core/src/data-module/subscription-store/subscriptionStore.ts

```diff
@@ -10,6 +10,11 @@ import type {
 } from '../types';
 import type { CacheSettings } from '../data-cache/types';
 
+function isQueryWithRequestSettings<Query extends DataStreamQuery>(
```

Comment:
Nit: Remove duplication of types in this file. There's multiple instances of `Query | (Query & { requestSettings: { refreshRate?: number } }` worth cleaning up.